components/claim-mgt/org.wso2.carbon.identity.claim.metadata.mgt/src/main/java/org/wso2/carbon/identity/claim/metadata/mgt/SystemDefaultClaimMetadataManager.java

```diff
@@ -0,0 +1,277 @@
+
+        if (externalClaimDialectURI == null || StringUtils.isBlank(externalClaimDialectURI) || claimURI == null ||
+                StringUtils.isBlank(claimURI)) {
+            return null;
+        }
+
+        if (externalClaimDialectURI.equals(LOCAL_CLAIM_DIALECT_URI)) {
+            throw new ClaimMetadataException("Invalid external claim dialect URI: " + externalClaimDialectURI);
+        }
+
+        return claims.getOrDefault(externalClaimDialectURI, Collections.emptyList()).stream()
+                .filter(claim -> claimURI.equals(claim.getClaimURI()))
+                .map(ExternalClaim.class::cast)
+                .findFirst()
+                .orElse(null);
+    }
+
+    @Override
+    public List<Claim> getMappedExternalClaims(String localClaimURI, int tenantId) throws ClaimMetadataException {
+
+        if (localClaimURI == null || StringUtils.isBlank(localClaimURI)) {
+            return null;
+        }
+
+        List<Claim> mappedExternalClaims = new ArrayList<>();
+        for (Map.Entry<String, List<Claim>> entry : claims.entrySet()) {
+            if (LOCAL_CLAIM_DIALECT_URI.equals(entry.getKey())) {
+                continue;
+            }
+            List<Claim> externalClaims = entry.getValue().stream()
+                    .map(ExternalClaim.class::cast)
+                    .filter(claim -> localClaimURI.equals(claim.getMappedLocalClaim()))
+                    .map(Claim.class::cast)
+                    .collect(Collectors.toList());
+            mappedExternalClaims.addAll(externalClaims);
+        }
+        return mappedExternalClaims;
+    }
+
+    @Override
+    public boolean isMappedLocalClaim(String localClaimURI, int tenantId) throws ClaimMetadataException {
+
+        if (localClaimURI == null || StringUtils.isBlank(localClaimURI)) {
+            return false;
+        }
+
+        for (Map.Entry<String, List<Claim>> entry : claims.entrySet()) {
+            if (LOCAL_CLAIM_DIALECT_URI.equals(entry.getKey())) {
+                continue;
+            }
+            List<Claim> externalClaims = entry.getValue().stream()
+                    .map(ExternalClaim.class::cast)
+                    .filter(claim -> localClaimURI.equals(claim.getMappedLocalClaim()))
+                    .collect(Collectors.toList());
+            if (!externalClaims.isEmpty()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public boolean isLocalClaimMappedWithinDialect(String mappedLocalClaim, String externalClaimDialectURI, int tenantId) throws ClaimMetadataException {
+
+        if (externalClaimDialectURI == null || StringUtils.isBlank(externalClaimDialectURI)
+                || !claims.containsKey(externalClaimDialectURI)) {
+            return false;
+        }
+        if (mappedLocalClaim == null || StringUtils.isBlank(mappedLocalClaim)) {
```

Comment:
refer to previous comment

Reply:
Fixed with ecc4327051a74cf340ca167fc5026149147e4caf